gpkit/small_classes.py

```diff
@@ -247,9 +244,10 @@ def __add__(self, other):
             return self.__class__({key: val+other
                                    for (key, val) in self.items()})
         elif isinstance(other, dict):
-            keys = set(self).union(other)
-            sums = {key: self.get(key, 0) + other.get(key, 0) for key in keys}
-            return self.__class__(sums)
+            sums = self.copy()
+            for key, value in other.items():
+                sums[key] = value + sums.get(key, 0)
+            return sums
```

Comment:
I tried my idea of implementing `copy()` for `HashVector`, feel free to drop that commit if you don't like it (eb37cd69a3aede19be13f9578fead492a9c83d01)